### PR TITLE
cinny-desktop: 4.0.3 -> 4.1.0, use hashRouter

### LINKED ISSUE #334631: `cinny-desktop` : Failed to load.
### Describe the bug
`cinny-desktop` : Fails to load after logging in
"Failed to load. WebAssembly.Module doesn't parse at byte 0: module doesn't start with '\0asm'"

### Steps To Reproduce
1. Install the `cinny-desktop` package
2. Log into a Matrix account
3. App fails to start with error mentioned before

### Expected behavior
-

### Screenshots
Screenshot_20240814_160359

### Additional context
-

### Notify maintainers

@Qyriad 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.3, NixOS, 24.11 (Vicuna), 24.11.20240809.5e0ca22`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.24.0pre20240723_fb450de2`
 - nixpkgs: `/nix/store/0jr2kk95c34c0b6yxi75q4fqgb43kqkm-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ryand56**
## Description of changes

Follow up to #331078, I was unable to rebase on top of current master due to many thousands of merge conflicts.
Updates cinny-desktop to 4.1.0, see changes: https://github.com/cinnyapp/cinny-desktop/releases/tag/v4.1.0
Closes #334631 and #329859.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Qyriad — APPROVED** [review]

**Comment by Guanran928** [comment]
Result of `nixpkgs-review pr 334805` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cinny-desktop</li>
  </ul>
</details>

**Comment by Guanran928** [comment]
I'm getting this error when I start cinny-desktop?

image

**Review by kfiven — CHANGES_REQUESTED** [review]

**Comment by ryand56** [comment]
#334638 is merged now, should we wait on this?

**Comment by emilazy** [comment]
Please don’t hold off on continuing to maintain and update these packages; that’s completely fine. The point of the libolm PR is just to offer users a warning about the current state of the cryptography library, and I fully expect that many will make their own threat assessment and choose to override it to continue using their preferred clients, which is totally okay. We don’t have any plans to deprecate or remove these clients given that the ecosystem is still in flux and working on migrating to other options. I’m sorry for the inconvenience to users of these packages, though; sadly `knownVulnerabilities` is the only real way we have to inform users about security risks they should be aware of.

**Comment by ryand56** [comment]
> Please don’t hold off on continuing to maintain and update these packages; that’s completely fine. The point of the libolm PR is just to offer users a warning about the current state of the cryptography library, and I fully expect that many will make their own threat assessment and choose to override it to continue using their preferred clients, which is totally okay. We don’t have any plans to deprecate or remove these clients given that the ecosystem is still in flux and working on migrating to other options. I’m sorry for the inconvenience to users of these packages, though; sadly `knownVulnerabilities` is the only real way we have to inform users about security risks they should be aware of.

Ah, thanks for clarifying that :)

**Review by Qyriad — APPROVED** [review]
cinny-desktop does fail to build on macOS, but unrelated to this PR, and this does fix the issue on Linux

**Comment by ryand56** [comment]
> cinny-desktop does fail to build on macOS, but unrelated to this PR, and this does fix the issue on Linux

Yes, it's related to this: #335334

File: pkgs/by-name/ci/cinny-desktop/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/ci/cinny-desktop/package.nix:1]: Please rebase and fix up this change

**Review by SuperSandro2000 — COMMENTED** [review]
Don't we also need to update web? tauri got reverted back to 1 on master.

**Comment by ryand56** [comment]
> Don't we also need to update web? tauri got reverted back to 1 on master.

Web was updated in #332619

**Comment by Qyriad** [comment]
It's now failing to launch on Wayland (curiously, even with `WAYLAND_DISPLAY` unset):
```
** (cinny:1305176): WARNING **: 12:11:56.501: webkit_settings_set_enable_offline_web_application_cache is deprecated and does nothing.
Gtk-Message: 12:11:57.056: Failed to load module "appmenu-gtk-module": 'gtk_module_display_init': /run/current-system/sw/lib/gtk-3.0/modules/libcolorreload-gtk-module.so: undefined symbol: gtk_module_display_init
Gtk-Message: 12:11:57.186: Failed to load module "appmenu-gtk-module": 'gtk_module_display_init': /run/current-system/sw/lib/gtk-3.0/modules/libcolorreload-gtk-module.so: undefined symbol: gtk_module_display_init
Gdk-Message: 12:11:57.639: Error 71 (Protocol error) dispatching to Wayland display.
```

**Comment by ryand56** [comment]
> It's now failing to launch on Wayland (curiously, even with `WAYLAND_DISPLAY` unset):
> 
> ```
> ** (cinny:1305176): WARNING **: 12:11:56.501: webkit_settings_set_enable_offline_web_application_cache is deprecated and does nothing.
> Gtk-Message: 12:11:57.056: Failed to load module "appmenu-gtk-module": 'gtk_module_display_init': /run/current-system/sw/lib/gtk-3.0/modules/libcolorreload-gtk-module.so: undefined symbol: gtk_module_display_init
> Gtk-Message: 12:11:57.186: Failed to load module "appmenu-gtk-module": 'gtk_module_display_init': /run/current-system/sw/lib/gtk-3.0/modules/libcolorreload-gtk-module.so: undefined symbol: gtk_module_display_init
> Gdk-Message: 12:11:57.639: Error 71 (Protocol error) dispatching to Wayland display.
> ```

Can't reproduce, launches just fine for me.

**Comment by Qyriad** [comment]
I'm getting some launch failure or another on KWin and Weston, Intel or Nvidia, `GDK_BACKEND=x11` or Wayland. What compositor and graphics driver are you on?

**Review by Qyriad — APPROVED** [review]
Weird, but probably just an issue with my setup.

**Comment by zincentimeter** [comment]
> I'm getting this error when I start cinny-desktop? image

I have the same problem. Want to add that terminal gives the following msg related to olm. 
The author said that upgrading to 4.1.0 will fix this. Hope this get merged ASAP.
```
** (cinny:316869): WARNING **: 11:11:57.274: webkit_settings_set_enable_offline_web_application_cache is deprecated and does nothing.
Asset `home/olm.wasm` not found; fallback to home/olm.wasm.html
Asset `home/olm.wasm` not found; fallback to home/olm.wasm.html
```

**Pull request merged**